test/models/Reads/DirectorOrPartnerDetailTypeItemReadsSpec.scala

```diff
@@ -0,0 +1,163 @@
+        Json.obj("lines" -> JsArray(Seq(JsString("line1"),JsString("line2"))),
+          "country" -> JsObject(Map("name" -> JsString("IT")))))
+
+      val directors = JsArray(Seq(director,director))
+
+      "We have director user details" when {
+        "We have a list of Directors" in {
+          val result = directors.as[Seq[DirectorOrPartnerDetailTypeItem]](DirectorOrPartnerDetailTypeItem.apiReads)
+
+          result.head.lastName mustBe directorSample.lastName
+        }
+
+        "We have a sequence id" in {
+          val result = directors.as[Seq[DirectorOrPartnerDetailTypeItem]](DirectorOrPartnerDetailTypeItem.apiReads)
+          result.head.sequenceId mustBe "0"
+        }
+
+
+        "We have a first name" in {
+          val result = directors.as[Seq[DirectorOrPartnerDetailTypeItem]](DirectorOrPartnerDetailTypeItem.apiReads)
+
+          result.head.firstName mustBe directorSample.firstName
+        }
+
+        "We have a last name" in {
+          val result = directors.as[Seq[DirectorOrPartnerDetailTypeItem]](DirectorOrPartnerDetailTypeItem.apiReads)
+
+          result.head.lastName mustBe directorSample.lastName
+        }
+
+        "We have a middle name " in {
+          val result = directors.as[Seq[DirectorOrPartnerDetailTypeItem]](DirectorOrPartnerDetailTypeItem.apiReads)
+
+          result.head.middleName mustBe directorSample.middleName
+        }
+
+        "We have a date of birth" in {
+          val result = directors.as[Seq[DirectorOrPartnerDetailTypeItem]](DirectorOrPartnerDetailTypeItem.apiReads)
+
+          result.head.dateOfBirth mustBe directorSample.dateOfBirth
+        }
+
+        "We don't have Title" in {
+          val result = directors.as[Seq[DirectorOrPartnerDetailTypeItem]](DirectorOrPartnerDetailTypeItem.apiReads)
+
+          result.head.title mustBe None
+        }
+      }
+
+      "We have director NINO details" when {
+        "We have a director nino" in {
+          val result = directors.as[Seq[DirectorOrPartnerDetailTypeItem]](DirectorOrPartnerDetailTypeItem.apiReads)
+
+          result.head.referenceOrNino mustBe directorSample.referenceOrNino
+        }
+
+        "We don't have a nino" in {
+          val directorsNoNino = directors.value :+ (director + ("directorNino" -> Json.obj("hasNino" -> JsBoolean(false))))
+
+          val result = JsArray(directorsNoNino).as[Seq[DirectorOrPartnerDetailTypeItem]](DirectorOrPartnerDetailTypeItem.apiReads)
+
+          result.last.referenceOrNino mustBe None
+        }
+
+        "We have a reason for not having nino" in {
+          val directorsNoNino = directors.value :+ (director + ("directorNino" -> Json.obj("reason" -> JsString("he can't find it"))))
+
+          val result = JsArray(directorsNoNino).as[Seq[DirectorOrPartnerDetailTypeItem]](DirectorOrPartnerDetailTypeItem.apiReads)
+
+          result.last.noNinoReason mustBe directorSample.noNinoReason
+        }
+      }
+
```

Comment:
Is there a case to require noNinoReason if nino is passed? Perhaps also the converse?